### PR TITLE
Fix trying to detach a float number in LNNP

### PR DESCRIPTION
**Pull request by RaulPPelaez**
If the user trains with only energy or only forces in some situations the LNNP class will try to call detach on a float where it expects a tensor.
I changed the loss computation so it initializes the losses on both energy and force as a torch.tensor.

### CODE REVIEW

**Comment by RaulPPelaez** [comment]
@raimis could you please review?

**Review by stefdoerr — APPROVED** [review]
I am not sure how optimized pytorch is but it might be slightly better to do `torch.zeros(1, device=self.device)`. In theory the above code will copy the value to the GPU while the other one might set it directly on the device. It's *extremely* insignificant performance-wise for a single number but I assume there is no reason not to do it :P Up to you

**Comment by stefdoerr** [comment]
Sorry scratch that, torch.zeros makes an array, torch.tensor makes a single float. Ignore my comment
